specs/enable-selinux.adoc

```diff
@@ -0,0 +1,197 @@
+So that the resource can be used with the restricted gated privilege.
+The file or directory created in a directory should also acquire same context.
+
+When the system runs SELinux in permissive mode, users are able to
+label files incorrectly. Files created with SELinux in permissive mode are not
+labeled correctly while files created while SELinux is disabled are not labeled
+at all. This behavior causes problems when changing to enforcing mode.
+Also there are some application that know about SELinux status can change their
+behavior when selinux is in permissive mode.
+
+== Use Cases
+
+* Enable SELinux security system for tendrl
+* Retain the existing security measures in an already enabled SELinux system
+
+== Proposed change
+
+* Running a tendrl related services in a Specific Security Context
+  Tendrl can have a selinux policies for all tendrl related services
+  Each tendrl proces can be assigned to an unique security context which,
+  just like with the user under which the process runs, helps Linux in
+  identifying what the application should and shouldn't be allowed to do.
+
+* Marking only the required type as permissive
+  In addition to have SELinux policies for tendrl, it would better to mark
+  only the required type as permissive initially; instead of making the
+  entire system permissive, tendrl could make only the purticular domain as permissive.
+  Over the release iterations, testing various test cases, SELinux can be
+  compleatly enforced. Because it's inevitable to completely block any new
+  avc errors which might come on any new updates to tendrl.
+
+=== Alternatives
+
+None
+
+=== Data model impact:
+
+None
+
+=== Impacted Modules:
+
+==== Tendrl API impact:
+
+SELinux policy files will be added in to this tendrl-api module.
+This set of policies will be used at tendrl server.
+
+Sample Policy:
+policy_module(tendrl, 1.0.0)
+
+########################################
+#
+# Declarations
+#
+
+type tendrl_t;
+type tendrl_exec_t;
+init_daemon_domain(tendrl_t, tendrl_exec_t)
+
+type tendrl_conf_t;
+files_config_file(tendrl_conf_t)
+
+type tendrl_log_t;
+logging_log_file(tendrl_log_t)
+
+type tendrl_var_lib_t;
+files_type(tendrl_var_lib_t)
+
+type tendrl_var_run_t;
+files_pid_file(tendrl_var_run_t)
+
+type tendrl_unit_file_t;
+systemd_unit_file(tendrl_unit_file_t)
+
+type tendrl_custom_port_t;
+corenet_port(tendrl_custom_port_t)
+
+permissive tendrl_t;
+
+########################################
+#
+# tendrl local policy
+#
+allow tendrl_t self:capability { sys_rawio sys_admin net_admin };
+allow tendrl_t self:fifo_file rw_fifo_file_perms;
+allow tendrl_t self:unix_stream_socket create_stream_socket_perms;
+allow tendrl_t self:tcp_socket { accept listen };
+- - -
+optional_policy(`
+          unconfined_domain(tendrl_t)
+')
+
+==== Tendrl commons impact:
+
+SELinux policy files will be added in to this tendrl-commons module.
+This will be used for every nodes participating in the tendrl.
```

Comment:
Does this include the storage nodes (gluster nodes) as well ?

Reply:
Yes, Initially we will be having a common policy for tendrl nodes and server.